src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DefaultDockerClient.java

```diff
@@ -150,6 +152,42 @@ public void pullImage(Image image) throws DockerServiceUnavailableException, Inv
         }
     }
 
+    /**
+     * Check if an image exists locally.
+     *
+     * @param image image to check locally
+     * @throws DockerServiceUnavailableException   an error that can be potentially fixed through retries
+     * @throws UserNotAuthorizedForDockerException when current user is not authorized to use docker
+     * @throws DockerImageQueryException unexpected error
+     */
+    public boolean imageExistsLocally(Image image) throws DockerServiceUnavailableException,
+            UserNotAuthorizedForDockerException, DockerImageQueryException {
+        CliResponse response = runDockerCmd(String.format("docker images -q %s", image.getImageFullName()));
+
+        Optional<UserNotAuthorizedForDockerException> userAuthorizationError = checkUserAuthorizationError(response);
+        if (userAuthorizationError.isPresent()) {
+            throw userAuthorizationError.get();
+        }
+
+        if (response.exit.isPresent()) {
+            if (response.exit.get() == 0) {
+                return StringUtils.isNotBlank(response.getOut());
+            } else {
+                if (response.getOut().contains("Service Unavailable")) {
+                    // This error can be thrown when disconnected/issue with docker cloud service, or when the docker
+                    // engine has issues or proxy config is bad etc. Not entirely reliable to determine retry behavior
+                    throw new DockerServiceUnavailableException(
```

Comment:
why does querying the local images need an internet connection?

Reply:
would using `docker image inspect` help in not using the network?

Reply:
+1. Do we know that it ever tries to contact the docker registry/service?